pkg/flowaggregator/flowaggregator.go

```diff
@@ -179,26 +197,26 @@ func NewFlowAggregator(
 	registry := ipfix.NewIPFIXRegistry()
 	registry.LoadRegistry()
 	fa := &flowAggregator{
-		externalFlowCollectorAddr,
-		externalFlowCollectorProto,
-		aggregatorTransportProtocol,
-		nil,
-		nil,
-		exportInterval,
-		nil,
-		0,
-		0,
-		registry,
-		ipfixentities.NewSet(false),
-		flowAggregatorAddress,
-		k8sClient,
-		observationDomainID,
+		externalFlowCollectorAddr:   externalFlowCollectorAddr,
+		externalFlowCollectorProto:  externalFlowCollectorProto,
+		aggregatorTransportProtocol: aggregatorTransportProtocol,
+		recordMetadataMap:           make(map[ipfixintermediate.FlowKey]recordMetadata, 0),
+		activeExpiryPriorityQueue:   make(TimeToExpirePriorityQueue, 0),
+		inactiveExpiryPriorityQueue: make(TimeToExpirePriorityQueue, 0),
```

Comment:
My first thought is that I am not sure why we need 2 different priority queues? Wouldn't we have less code and less redundancy with a single priority queue? Each item can have an active expiration time and an inactive expiration time, with the inactive timeout being reset every time a new record is received for the flow. I could be missing some context, so would like to hear your thoughts.

Reply:
Here I went with the model of maintaining two goroutines one for active expiry and another one for inactive/idle expiry. Each goroutine focusing on a separate priority queue with only one type of expiration time makes implementation simpler.

With both active and inactive expiration time in single priority queue, I presume we have to use a min(active expiry, inactive expiry) for each element to maintain the heap rather than picking one expiry time. In this case, we probably can use one goroutine. To keep things simple in terms of implementation, I went with former.